### PR TITLE
update to include new features

### PR DESCRIPTION
**Pull request by snehara99**
updated automatic reconnection, special characters, and message encoding. 
NOTE: tcp still needs to be added

### CODE REVIEW

File: docs/embedded/serial-monitor.md [hunk 1]
```diff
@@ -25,8 +25,9 @@ The Serial Monitor allows users to configure, monitor, and communicate with seri
 ## Capabilities
 
 - **Monitor a serial port**: Choose the **Start Monitoring** or **Stop Monitoring** button to control whether to monitor data coming from the port.
-- **Send data to a serial port**: Enter text into the text field at the bottom of the view. Use the **Enter** key or choose the **Send Message** button to send the data.
```
Comment (gcampbell-msft) [inline docs/embedded/serial-monitor.md:28]: Why did we remove the bit about using the "Enter" key?

**Review by gcampbell-msft — APPROVED** [review]

File: docs/embedded/serial-monitor.md (hunk 1 above)
Comment (snehara99) [reply docs/embedded/serial-monitor.md:28]: Oh I didn't realize we ever had that feature. I can add it back

Comment (gcampbell-msft) [reply docs/embedded/serial-monitor.md:28]: I just think it's useful to know for accessibility. Basically, if you're typing in the input box and hit enter, it should send.

Comment (snehara99) [reply docs/embedded/serial-monitor.md:28]: fair, I reverted it here. I'll update the code to allow enter too

Comment (gcampbell-msft) [reply docs/embedded/serial-monitor.md:28]: It should already be there, depending on your changes.

File: docs/embedded/serial-monitor.md [hunk 2]
```diff
@@ -35,6 +36,8 @@ The Serial Monitor allows users to configure, monitor, and communicate with seri
...
 | **Baud Rate** | Frequency at which the monitor attempts to communicate with the connected device | Use the **Baud Rate** dropdown | 300, 1200, 2400, 4800, 9600, 19200, 38400, 57600, 74880, 115200, 230000, 250000 |
-| **Line ending** | The line ending to use in messages sent to the connected device | Use the **Line Ending** dropdown | None, LF, CR, CRLF |
+| **Line Ending** | The line ending to use in messages sent to the connected device | Use the **Line Ending** dropdown | None, LF, CR, CRLF |
 | **Timestamp** | Adds timestamps to the output of data received from the connected port | Use the **timestamp** toggle button | On/Off |
-| **Auto-scroll** | Whether to auto-scroll new content that comes from the connected port | Use the **auto-scroll** toggle button | On/Off |
+| **Autoscroll** | Whether to auto-scroll new content that comes from the connected port | Use the **autoscroll** toggle button | On/Off |
+! **Automatic Reconnection** | Allows for automatic reconnection and monitoring of disconnected selected ports | Use the **automatic reconnection** toggle button | On/Off|
```
Comment (Jak-MS) [inline docs/embedded/serial-monitor.md:42]:
```suggestion
| **Automatic Reconnection** | Allows for automatic reconnection and monitoring of disconnected selected ports | Use the **automatic reconnection** toggle button | On/Off|
```
fixing table

**Review by Jak-MS — APPROVED** [review]

**Pull request merged**